pyOCD/tools/gdb_server.py

```diff
@@ -137,6 +141,81 @@ def server_listening(self, server):
             print >> sys.stderr, self.echo_msg
             sys.stderr.flush()
 
+    def disable_logging(self):
+        logging.getLogger().setLevel(logging.FATAL)
+
+    def list_boards(self):
+        self.disable_logging()
+
+        try:
+            all_mbeds = MbedBoard.getAllConnectedBoards(close=True, blocking=False)
+            status = 0
+            error = ""
+        except Exception as e:
+            all_mbeds = []
+            status = 1
+            error = str(e)
+            if not self.args.output_json:
+                raise
+
+        if self.args.output_json:
+            boards = []
+            obj = {
+                'pyocd_version' : __version__,
+                'version' : { 'major' : 1, 'minor' : 0 },
+                'status' : status,
+                'boards' : boards,
+                }
+
+            if status != 0:
+                obj['error'] = error
+
+            for mbed in all_mbeds:
+                d = {
+                    'unique_id' : mbed.unique_id,
+                    'info' : mbed.getInfo(),
+                    'board_name' : mbed.getBoardName(),
+                    'target' : mbed.getTargetType(),
+                    'vendor_name' : mbed.interface.vendor_name,
+                    'product_name' : mbed.interface.product_name,
+                    }
+                boards.append(d)
+
+            print json.dumps(obj, indent=4) #, sys.stdout)
+        else:
+            index = 0
+            if len(all_mbeds) > 0:
+                for mbed in all_mbeds:
+                    print("%d => %s boardId => %s" % (index, mbed.getInfo().encode('ascii', 'ignore'), mbed.unique_id))
+                    index += 1
+            else:
+                print("No available boards are connected")
+
+    def list_targets(self):
+        self.disable_logging()
```

Comment:
list_targets is only planned on being used in this file right?  If not you might want to restore the logging state.